inst/include/dust/dust.hpp

```diff
@@ -308,6 +310,19 @@ class Dust {
     }
   }
 
+  // Used for copy of state into another block of memory on the device (used
+  // for history saving)
+  void state(dust::device_array<real_t>& device_state, const size_t dst_offset,
```

Comment:
I may fail to find where this is called or connect my thoughts up here, but:

Is there a good reason this overloads the state name? It seems more like something that is more connected to the device logic and better suited to being a private method with a different name

Reply:
Going to leave these for now. Once we've done #227 I think we should be able to combine the two filter functions with a template by using this style